crates/wash-lib/src/build/component.rs

```diff
@@ -315,6 +315,33 @@ async fn build_tinygo_component(
             ".",
         ],
         WasmTarget::WasiP2 => {
+            // If we are using tinygo without a custom path, check the tinygo
+            // version to make sure that it can support WasiP2, aka we're using
+            // 0.33.0+
+            if tinygo_config.tinygo_path.is_none() {
+                // Run `tinygo version` and parse the output for version
+                let output = tokio::process::Command::new("tinygo")
+                    .arg("version")
+                    .output()
+                    .await
+                    .context("failed to get tinygo version")?
+                    .stdout;
+
+                let version = std::str::from_utf8(&output)
+                    .context("failed to parse tinygo version")?
+                    .split(' ')
+                    .filter_map(|s| semver::Version::parse(s).ok())
+                    .next()
+                    .unwrap_or_else(|| semver::Version::new(0, 0, 0));
```

Comment:
```suggestion
                    .unwrap_or_else(|| semver::Version::new(0, 0, 0));
```

Is this possible? should we just error if we can't figure out a version?